### PR TITLE
Link assisters and supervisors to actionable tab

### PR DESCRIPTION
**Pull request by bensheldon**
- also added more access control for based on AidApplication status

### CODE REVIEW

File: app/views/admin/aid_applications/index.html.erb [hunk 1]
```diff
@@ -22,7 +22,7 @@
         <% @aid_applications.each do |aid_application| %>
           <tr id="<%= dom_id(aid_application) %>">
             <td><%= aid_application.id %></td>
-            <td><%= link_to aid_application.application_number, edit_organization_aid_application_eligibility_path(aid_application.organization, aid_application) %>
+            <td><%= link_to aid_application.application_number, organization_aid_application_path(aid_application.organization, aid_application) %>
```
Comment (wschaefer) [inline app/views/admin/aid_applications/index.html.erb:25]: so admins will go to the same view as assisters if they click an application? that seems like a reasonable default.

File: app/controllers/aid_applications/base_controller.rb [hunk 2]
```diff
@@ -10,5 +10,29 @@ def current_aid_application
       @_current_aid_application ||= current_organization.aid_applications.find(params[:aid_application_id])
     end
     helper_method :current_aid_application
+
+    def ensure_eligible
```
Comment (wschaefer) [inline app/controllers/aid_applications/base_controller.rb:14]: these ensure_<status> methods weren't necessary for these stories, right? using the aid_applications#show to redirect to the right place was sufficient i think? it seems like reasonable behavior to restrict what tab a user can continue to based on the status, but i was just curious if there was a problem happening that prompted you to add this now?

File: app/views/admin/aid_applications/index.html.erb (hunk 1 above)
Comment (bensheldon) [reply app/views/admin/aid_applications/index.html.erb:25]: Yes. The story was a little under-specified about where admins should go. The Submission tab has a lot of info and it felt weird to bump the admin to action a supervisor might take.

Thinking more, maybe it should be slightly smarter, like bumping the admin to submission or finish.

File: app/controllers/aid_applications/base_controller.rb (hunk 2 above)
Comment (bensheldon) [reply app/controllers/aid_applications/base_controller.rb:14]: They weren't necessary for the stories. But given that I was pushing people to more places, I got a little paranoid that someone would be bumped along to the wrong place. So I did a sorta belt (redirect) -and -suspenders (access control) to ensure that we don't bump someone to the wrong place now or in the future.

**Comment by bensheldon** [comment]
@wschaefer thanks for looking at this. I also spaced on seeing all the test failures. Turns out some tests were relying on incomplete applications 👻

**Pull request merged**